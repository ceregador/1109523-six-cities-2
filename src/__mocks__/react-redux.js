export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
