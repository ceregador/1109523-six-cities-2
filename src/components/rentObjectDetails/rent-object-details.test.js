import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import RentObjectDetails from './rent-object-details.jsx';
import Renderer from 'react-test-renderer';
import PageHeader from '../pageHeader/page-header.jsx';
import Rating from '../rating/rating.jsx';
import ReviewList from '../reviewList/review-list.jsx';
import ReviewForm from '../reviewForm/review-form.jsx';
import OffersMap from '../offersMap/offers-map.jsx';
import CardList from '../cardList/card-list.jsx';

jest.mock(`../rating/rating.jsx`);
jest.mock(`../reviewList/review-list.jsx`);
jest.mock(`../reviewForm/review-form.jsx`);
jest.mock(`../offersMap/offers-map.jsx`);
jest.mock(`../cardList/card-list.jsx`);
jest.mock(`../pageHeader/page-header.jsx`);

it(`renders placeholder if offer is undefined and connects`, () => {
  const tree = Renderer
    .create(
        <RentObjectDetails
          isAuthorized={false}
          offer={null}
          addToFavorites={jest.fn()}
          fetchDataForHotel={jest.fn()}
          match={{params: {offerId: `1`}}}
          activeCity={null}
          cityOffers={[]}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({offers: [], cities: []});
  expect(mappedProps).toHaveProperty(`offer`);
  expect(mappedProps).toHaveProperty(`activeCity`);
  expect(mappedProps).toHaveProperty(`cityOffers`);
  expect(mappedProps).toHaveProperty(`isAuthorized`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`addToFavorites`);
  expect(mapDispatchToProps).toHaveProperty(`fetchDataForHotel`);

  expect(connectAdvanced).toHaveBeenCalledWith(RentObjectDetails);

  expect(tree).toMatchSnapshot();
});

it(`renders and connects correctly`, () => {
  const tree = Renderer
    .create(
        <RentObjectDetails
          isAuthorized={true}
          offer={{
            id: 1,
            name: ``,
            images: [],
            equipment: [],
            host: {
              avatarUrl: ``,
              name: ``,
              isPro: true},
            rating: 1,
            price: 1,
            type: ``,
            description: ``,
            bedroomsCount: 1,
            maxGuestsCount: 1,
            isPremium: false,
            isBookmarked: false
          }}
          addToFavorites={jest.fn()}
          fetchDataForHotel={jest.fn()}
          match={{params: {offerId: `1`}}}
          activeCity={{name: ``, coordinates: []}}
          cityOffers={[]}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({offers: [], cities: []});
  expect(mappedProps).toHaveProperty(`offer`);
  expect(mappedProps).toHaveProperty(`activeCity`);
  expect(mappedProps).toHaveProperty(`cityOffers`);
  expect(mappedProps).toHaveProperty(`isAuthorized`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`addToFavorites`);
  expect(mapDispatchToProps).toHaveProperty(`fetchDataForHotel`);

  expect(connectAdvanced).toHaveBeenCalledWith(RentObjectDetails);

  expect(PageHeader).toHaveBeenCalled();
  expect(Rating).toHaveBeenCalled();
  expect(ReviewList).toHaveBeenCalled();
  expect(ReviewForm).toHaveBeenCalled();
  expect(OffersMap).toHaveBeenCalled();
  expect(CardList).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
