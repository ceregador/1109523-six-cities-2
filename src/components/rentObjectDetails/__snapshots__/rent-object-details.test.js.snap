// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and connects correctly 1`] = `
<div
  className="page"
>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        />
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              
            </h1>
            <button
              className="property__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              1
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              1
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            />
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
    </section>
  </main>
</div>
`;

exports[`renders placeholder if offer is undefined and connects 1`] = `
<p>
  Loading...
</p>
`;
